File: src/starter.js
// === Clique Home Assignment Starter (Node.js) ===
// You may freely modify this file or start from scratch.

import fs from "fs";
import path from "path";

const DATA_PATH = "../data";

function loadCSV(filename) {
  const text = fs.readFileSync(path.join(DATA_PATH, filename), "utf8");
  const [headerLine, ...lines] = text.trim().split("\n");
  const headers = headerLine.split(",");
  return lines.map(line => {
    const values = line.split(",");
    const obj = {};
    headers.forEach((h, i) => (obj[h] = values[i]));
    return obj;
  });
}

function loadData() {
  const members = loadCSV("members.csv");
  const lastContacts = loadCSV("last_contacts.csv");
  const holidays = JSON.parse(fs.readFileSync(path.join(DATA_PATH, "holidays.json")));
  return { members, lastContacts, holidays };
}

function calculateDueMembers({ members, lastContacts, holidays }) {
  // TODO: Implement the assignment logic here
  return [];
}

function main() {
  const data = loadData();
  const result = calculateDueMembers(data);
  console.log("Members due for check-in:");
  console.log(result);
}

main();

File: clique_home_assignment/src/starter.py
import csv
import json
from datetime import datetime, timezone

# === Clique Home Assignment Starter (Python) ===
# You may freely modify this file or start from scratch.
# The data files are located in ../data/

DATA_PATH = "../data"

def load_data():
    with open(f"{DATA_PATH}/members.csv") as f:
        members = list(csv.DictReader(f))
    with open(f"{DATA_PATH}/last_contacts.csv") as f:
        last_contacts = list(csv.DictReader(f))
    with open(f"{DATA_PATH}/holidays.json") as f:
        holidays = json.load(f)
    return members, last_contacts, holidays

def calculate_due_members(members, last_contacts, holidays):
    # TODO: Implement logic from the assignment
    # 1. Determine who needs a check-in
    # 2. Compute priority_score
    # 3. Sort and return list
    return []

def main():
    members, last_contacts, holidays = load_data()
    results = calculate_due_members(members, last_contacts, holidays)
    print("Members due for check-in:")
    for m in results:
        print(m)

if __name__ == "__main__":
    main()
